### PR TITLE
Added max_sum_bst.py

### PR DESCRIPTION
**Pull request by 1227haran**
### Describe your change:

* [X] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [X] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [X] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by imSanko — CHANGES_REQUESTED** [review]
Visit the CONTRIBUTION.md for info about testing the Code before an PR

File: data_structures/binary_tree/maximum_sum_bst.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+class TreeNode:
```
Comment (cclauss) [inline data_structures/binary_tree/maximum_sum_bst.py:1]: Please make this a https://docs.python.org/3/library/dataclasses.html

**Comment by cclauss** [comment]
On your computer: `python3 -m doctest data_structures/binary_tree/maximum_sum_bst.py`

File: data_structures/binary_tree/maximum_sum_bst.py (hunk 1 above)
Comment (cclauss) [reply data_structures/binary_tree/maximum_sum_bst.py:9]: See https://github.com/TheAlgorithms/Python/blob/master/data_structures/binary_tree/binary_tree_traversals.py#L10 for a working example.

File: data_structures/binary_tree/maximum_sum_bst.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import sys
+from dataclasses import dataclass
+from typing import Optional
```
Comment (cclauss) [inline data_structures/binary_tree/maximum_sum_bst.py:3]:
```suggestion
```

File: data_structures/binary_tree/maximum_sum_bst.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+INT_MIN = -sys.maxsize + 1
+INT_MAX = sys.maxsize - 1
+
+
+@dataclass
+class TreeNode:
+    val: int = 0
+    left: Optional["TreeNode"] = None
+    right: Optional["TreeNode"] = None
```
Comment (cclauss) [inline data_structures/binary_tree/maximum_sum_bst.py:13]:
```suggestion
    left: TreeNode | None = None
    right: TreeNode | None = None
```

File: data_structures/binary_tree/maximum_sum_bst.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+
+@dataclass
+class TreeNode:
+    val: int = 0
+    left: TreeNode | None = None
+    right: TreeNode | None = None
+
+
+def max_sum_bst(root: TreeNode) -> int:
```
Comment (cclauss) [inline data_structures/binary_tree/maximum_sum_bst.py:17]:
```suggestion
def max_sum_bst(root: TreeNode | None) -> int:
```

File: data_structures/binary_tree/maximum_sum_bst.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    >>> t3.right.left = TreeNode(2)
+    >>> t3.right.right = TreeNode(5)
+    >>> t3.right.right.left = TreeNode(4)
+    >>> t3.right.right.right = TreeNode(6)
+    >>> print(max_sum_bst(t3))
+    20
+    """
+    ans: int = 0
+
+    def solver(node: TreeNode) -> tuple[bool, int, int, int]:
```
Comment (cclauss) [inline data_structures/binary_tree/maximum_sum_bst.py:49]:
```suggestion
    def solver(node: TreeNode | None) -> tuple[bool, int, int, int]:
```

**Review by cclauss — APPROVED** [review]
Nice!  Thanks for doing this.

**Pull request merged**